lib/sw-toolbox.js

```diff
@@ -32,7 +32,9 @@ self.addEventListener('install', function(event) {
   helpers.debug('preCache list: ' + (options.preCacheItems.join(', ') || '(none)'));
   event.waitUntil(
     helpers.openCache({cacheName: inactiveCache}).then(function(cache) {
-      return cache.addAll(options.preCacheItems);
+      return Promise.all(options.preCacheItems).then(function(items) {
```

Comment:
I believe you can omit the anonymous function and just write `return Promise.all(options.preCacheItems).then(cache.addAll);`